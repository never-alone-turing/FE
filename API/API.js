export const apiCall = (data) => {
  fetch("https://never-alone-backend.herokuapp.com/graphql", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ query: `
    mutation
      {
        updateCheckin( 
          input: {
            id: "1",
            response: "Answered"
          }
        ) {
          checkin{
            id
            response
            name
            userId
            groupId
            time
            window
          }
        }
      }`
    }),
  })
  .then(res => res.json())
  .then(res => console.log(res.data));
}
// export const apiCall = (data) => {
//   fetch('https://never-alone-backend.herokuapp.com/graphql', {
//     method: 'POST',
//     headers: { 'Content-Type': 'application/json' },
//     body: JSON.stringify({ query: `
//       {
//         allCheckins{
//           id
//         }
//       }`
//     }),
//   })
//   .then(res => res.json())
//   .then(res => console.log(res.data));
// }