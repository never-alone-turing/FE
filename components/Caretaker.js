import React from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";

const Caretaker = ({history}) => {
  return (
    <>
      <Text>Caretaker</Text>
      <TouchableOpacity style={styles.listItem} onPress={() => history.push('/')}>
        <View style={styles.listItemView}>
          <Text>Back</Text>
        </View>
      </TouchableOpacity>
    </>
  );
}

const styles = StyleSheet.create({
  listItem: {
    padding: 15,
    backgroundColor: '#f8f8f8',
    borderBottomWidth: 1,   
    borderColor: "#eee"
  },
  listItemView: {
    flexDirection: 'row'
  },
  listItemText: {
    fontSize: 18
  }
})

export default Caretaker;