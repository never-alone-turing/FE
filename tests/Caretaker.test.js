import React from 'react';
import Caretaker from '../components/Caretaker';
import renderer from 'react-test-renderer';
import DateTimePicker from '@react-native-community/datetimepicker';

jest.mock('@react-native-community/datetimepicker', () => jest.fn());
DateTimePicker.mockImplementation((props) => (
  <Input testID={props.testID} onFocus={() => props.onChange} />
));

import {fetcher} from '../API/API.js';

const mockReceiverAllTimers = jest.fn();
jest.mock('../API/API.js', async() => {
  return await jest.fn().mockImplementation(() => {
    return allCheckins = [{
      id: 1,
      response: "Answered",
      name: "test",
      time: new Date(0),
      window: new Date(0),
    }]
  })
})

// jest.mock('../API/API.js', () => jest.fn());
// fetcher.allTimers.mockImplementation( () => {
//   return allCheckins = [{
//     id: 1,
//     response: "Answered",
//     name: "test",
//     time: new Date(0),
//     window: new Date(0),
//   }]
// })
// jest.mock('../API/API.js', () => {
//   return fetcher.allTimers();
// })

// jest.mock("../node_modules/react-native/Libraries/LogBox/Data/LogBoxData.js", () => {
//     return {
//       handleException: (...args) => {
//         return;
//       },
//     };
//   })

describe('Caretaker', () => {
  it('should match the snapshot', async() => {
    const caretaker = await renderer.create(<Caretaker />).toJSON()
    expect(caretaker).toMatchSnapshot()
  })
})
// import React from 'react';
// import { render, fireEvent } from '@testing-library/react-native';
// import renderer from 'react-test-renderer';
// // import Caretaker from '../components/Caretaker';
// import Caretaker from '../components/Caretaker.js';
// import DateTimePicker from '@react-native-community/datetimepicker';

// jest.mock('@react-native-community/datetimepicker', () => jest.fn());
// DateTimePicker.mockImplementation((props) => (
//   <Input testID={props.testID} onFocus={() => props.onChange} />
// ));

// import {fetcher} from '../API/API.js';

// jest.mock('../API/API.js', () => jest.fn())

// // jest.mock('../API/API.js', () => jest.fn());
// // fetcher.mockImplementation((props) => (
// //   <Input testID={props.testID} onFocus={() => props.onChange} />
// // ));

// describe('Caretaker', () => {
//   let mockProps= {
//     navigation: {
//       navigate: jest.fn()
//     }
//   };
//   // it('should match the snapshot', () => {
//   //   const caretaker = renderer.create(<Caretaker/>).toJSON();
//   //   expect(caretaker).toMatchSnapshot()
//   // });
//   test('Render Caretaker Properly', () => {
//     const {debug} = render(<Caretaker navigation={mockProps.navigation}/>);
//     debug();
//   });
// })