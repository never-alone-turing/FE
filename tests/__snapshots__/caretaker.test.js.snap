// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Caretaker /> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#fff",
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
          text="Never Alone"
        >
          Never Alone
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "name": "Wake-up",
          "time": "7:00",
          "window": "0:15",
        },
        Object {
          "id": 2,
          "name": "Take meds",
          "time": "12:00",
          "window": "1:00",
        },
        Object {
          "id": 3,
          "name": "go to sleep",
          "time": "22:00",
          "window": "1:00",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#f8f8f8",
              "borderBottomWidth": 1,
              "borderColor": "#eee",
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  X
                </Text>
              </View>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Name: 
              Wake-up
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Time: 
              7:00
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Window: 
              0:15
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#f8f8f8",
              "borderBottomWidth": 1,
              "borderColor": "#eee",
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  X
                </Text>
              </View>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Name: 
              Take meds
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Time: 
              12:00
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Window: 
              1:00
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#f8f8f8",
              "borderBottomWidth": 1,
              "borderColor": "#eee",
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  X
                </Text>
              </View>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Name: 
              go to sleep
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Time: 
              22:00
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 18,
                }
              }
            >
              Window: 
              1:00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": "black",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginRight": 30,
              "marginTop": 50,
              "padding": 10,
              "textAlign": "center",
              "width": "auto",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "bold",
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Close Menu
          </Text>
        </View>
      </View>
    </View>
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Name"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 25,
            "height": 60,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Time"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 25,
            "height": 60,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Window"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 25,
            "height": 60,
            "padding": 8,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "black",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 30,
                  "paddingVertical": 1,
                  "textAlign": "center",
                }
              }
            >
              Add Item
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
